pkg/internal/session/session_test.go

```diff
@@ -40,3 +217,50 @@ var _ = Describe("Session operations", func() {
 		})
 	})
 })
+
+// helper function to mimic the server side reacting to the session object.
+func addSessionRefStatus(c *session.Client, sessionName string) func() {
+	done := false
+	go func() {
+		for {
+			if done {
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Why `100`? Is it significant? Also - why sleeping here?

Reply:
it's a while loop. It needs to take a break to not kill the cpu. :) 100 is semi random, but cli waits 1s pr check.

Reply:
But what are we waiting for here? My thinking is that every test relying on it will add `100ms` penalty which might not be necessary. Or am I missing something here?

Reply:
https://github.com/maistra/istio-workspace/blob/master/pkg/internal/session/session.go#L178

* The Client creates a Session object
* The Client starts a wait cycle of 1s*10 looking for the Status field 
* The Operator reacts to the Session object and creates the Status field (the function simulates this)
* The Client notices the Status field and returns

I couldn't find an easy way to make a 'callback' on Create/Update or similar, so both sides 'pull'. The 'wait' is that the Session object might not be created yet when this starts up, or it will at some point change. So it checks every 100ms to see if there is something to do.

Reply:
It doesn't matter if that is 100ms or 1ms, both are to slow to take part in the first Client check, which means anything <1s is fine.  The combined added time for these are 4s.

Reply:
We could add a time.Duration to our Options to lower the 1s wait time for tests tho..

Reply:
https://github.com/maistra/istio-workspace/pull/492/commits/72892c8a381cce63c781f8d2a443419c07b2255d

That makes it 0.04s instead of 4s